app/modules/wifi.c

```diff
@@ -652,6 +652,30 @@ static int wifi_station_getconfig_default(lua_State *L)
   return wifi_station_getconfig(L, true);
 }
 
+// Lua: wifi.sta.clearconfig()
+static int wifi_station_clear_config ( lua_State* L )
+{
+  struct station_config sta_conf;
+  bool auto_connect=true;
+  bool save_to_flash=true;
+
+  memset(sta_conf.ssid, 0, sizeof(sta_conf.ssid));
+  memset(sta_conf.password, 0, sizeof(sta_conf.password));
+  memset(sta_conf.bssid, 0, sizeof(sta_conf.bssid));
+  sta_conf.bssid_set=0;
+
+  wifi_station_disconnect();
+
+  bool config_success;
+  if(save_to_flash) config_success = wifi_station_set_config(&sta_conf);
```

Comment:
That was existing code - not changed for this PR. I'm all for applying the Boy Scout Rule, but think it makes it harder for reviewers to see what exactly is new.